src/v/iceberg/update_schema_action.cc

```diff
@@ -41,9 +41,13 @@ ss::future<action::action_outcome> update_schema_action::build_updates() && {
     const schema::id_t new_schema_id{highest_schema_id() + 1};
     new_schema_.schema_id = new_schema_id;
 
-    // TODO: when we support schema evolution, we'll need to assign IDs to only
-    // the new fields.
-    new_schema_.assign_fresh_ids();
+    // NOTE: assumes that
+    //   - the table's last_column_id increases monotonically
+    //   - carry-over fields in a compat-checked schema already have
+    //     IDs assigned
+    //   - those carry-over IDs are all strictly <= table.last
```

Comment:
Not suggesting we change even if it isn't the case, but just curious if these are the same invariants Iceberg's Java impl has?

Reply:
which part? (2) and (3) are implementation details viz. a) we update incoming fields w/ existing IDs and b) that the last column ID of the table is correct with respect to the columns therein

(1) is perhaps expressed here: 

https://github.com/apache/iceberg/blob/8e456aeeabd0a40b23864edadd622b45cb44572c/core/src/main/java/org/apache/iceberg/TableMetadata.java#L577-L581

Which matches the new logic in table_update_applier.